doc/modules/calibration.rst

```diff
@@ -205,3 +205,124 @@ a similar decrease in log-loss.
     .. [5] On the combination of forecast probabilities for
            consecutive precipitation periods. Wea. Forecasting, 5, 640–650.,
            Wilks, D. S., 1990a
+
+
+.. _decision_threshold_calibration:
+
+==============================
+Decision Threshold calibration
+==============================
+
+.. currentmodule:: sklearn.calibration
+
+Often Machine Learning classifiers base their
+predictions on real-valued decision functions or probability estimates that
+carry the inherited biases of their models. Additionally when using a machine
+learning model the evaluation criteria can differ from the optimisation
+objectives used by the model during training.
+
+When predicting between two classes it is commonly advised that an appropriate
+decision threshold is estimated based on some cutoff criteria rather than
+arbitrarily using the midpoint of the space of possible values. Estimating a
+decision threshold for a specific use case can help to increase the overall
+accuracy of the model and provide better handling for sensitive classes.
+
+.. currentmodule:: sklearn.calibration
+
+:class:`CutoffClassifier` can be used as a wrapper around a model for binary
+classification to help obtain a more appropriate decision threshold and use it
+for predicting new samples.
+
+Usage
+-----
+
+To use the :class:`CutoffClassifier` you need to provide an estimator that has
+a ``decision_function`` or a ``predict_proba`` method. The ``method``
+parameter controls whether the first will be preferred over the second if both
+are available.
+
+The wrapped estimator can be pre-trained, in which case ``cv = 'prefit'``, or
+not. If the classifier is not trained then a cross-validation loop specified by
+the parameter ``cv`` can be used to obtain a decision threshold by averaging
```

Comment:
Is this the right thing / a sensible thing to do? Do we have references for this? For CallibratedClassifierCV we just keep all the models and average them. We could do the same here. That might make more sense, but I don't know of any literature. Have you done any experiments?

Reply:
What experiments do you have in mind? Not more than running evaluating classifiers prediction accuracy, tpr, tnr given the input parameters of the cutoffclassifier. Literature wise I didn't find anythings related to cross validation and cutoff point. But maybe I haven't searched enough. 

> For CallibratedClassifierCV we just keep all the models and average them

do suggest to instead of keeping the decision threshold to  keep all the underlying trained models and combine / average their predictions ? what would be the combining criteria for the predictions in this case? just a mean of the predictions?

Reply:
Right, it would be less obvious how to combine, but voting would be possible. Maybe some experiments that confirm that the current implementation is a sensible thing to do and works in practice? I.e. averaging the thresholds makes it better, not worse, than a single hold-out?

Reply:
> I.e. averaging the thresholds makes it better, not worse, than a single hold-out? 

I've seen this in practice. My understanding was that the cv approach is also a way to obtain a threshold on your training data without worrying that it's overfit. Didn't expect it to necessarily improve the threshold or significantly. 

So what would be the purpose of keeping the underlying models? allowing the user to combine them whatever way they want?

Reply:
The point is that I haven't seen it in practice and I don't know of any write-up that says it's happening in practice so I'd like to be convinced ;)
And yes, it's a way to avoid overfitting. But another way would be to use a single split, and using a single split "obviously works" in some sense, but it's not entirely clear to me that averaging leads to a meaningful result.